.github/workflows/android.yml

```diff
@@ -18,21 +18,19 @@ on:
         description: 'release type'
         required: true
         default: 'qa'
-      targetbranch:
-        description: 'target deployment branch'
-        required: false
-        default: 'staging'
 
 # A workflow run is made up of one or more jobs that can run sequentially or in parallel
 jobs:
   # This workflow contains a single job called "build"
   build:
     # The type of runner that the job will run on
     runs-on: ubuntu-latest
-    # Build runs if triggered by merging pull request or by push with 'release' flag and without skipping CI :
+    # Build runs if triggered by merging pull request to master or by push with 'release <target_branch>' flag and without skipping CI :
     if: |
-      ${{ github.event_name == 'pull_request' && github.event.pull_request.merged == true ||
-      github.event_name == 'push' && !contains(github.event.head_commit.message,'release:') && !contains(github.event.head_commit.message,'skip ci') }}
+      ${{ github.event_name == 'pull_request' && github.event.pull_request.merged == true && endsWith(github.ref, '/master') && github.event.head_commit.message =~ .*\d{4} ||
+      github.event_name == 'push' && endsWith(github.ref, '/staging') && contains(github.event.head_commit.message, 'release qa') ||
+      github.event_name == 'push' && endsWith(github.ref, '/next') && contains(github.event.head_commit.message, 'release prod') ||
+      && !contains(github.event.head_commit.message,'skip ci') }}
```

Comment:
'skip ci' check  could be removed as we're checking for each branch that commit contains release **for this branch exactly**